ui-core/src/components/inputs/TolerancePicker.tsx

```diff
@@ -0,0 +1,129 @@
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { minusToleranceIndex, plusToleranceIndex } = useMemo(() => {
+    const minusToleranceIndex = toleranceRanges.findIndex(
+      (range) => range.value === minusTolerance
+    );
+    const plusToleranceIndex = toleranceRanges.findIndex((range) => range.value === plusTolerance);
+    if (minusToleranceIndex < 0 || plusToleranceIndex < 0)
+      setWarningStatus({
+        status: 'warning',
+        message: "One of the values isn't included in the ranges provided.",
+      });
+    return { minusToleranceIndex, plusToleranceIndex };
+  }, [minusTolerance, plusTolerance, toleranceRanges]);
+
+  useEffect(() => {
+    setInputValue(formatToleranceValue(minusToleranceIndex, plusToleranceIndex));
+  }, [minusTolerance, plusTolerance]);
+
+  return (
+    <div className="tolerance-picker">
+      <div>
+        <Input
+          {...inputProps}
+          value={inputValue}
+          // Event not used for now
+          onChange={(e) => {
+            setInputValue(e.target.value);
+            setShowPicker(true);
+          }}
+          statusWithMessage={warningStatus}
+          readOnly
+          onClick={() => setShowPicker(!showPicker)}
+          type="text"
+          ref={inputRef}
+        />
+      </div>
+
+      <Modal inputRef={inputRef} isOpen={showPicker} onClose={() => setShowPicker(false)}>
+        <div className="time-tolerance">
+          <div className="time-picker-section">
+            <div className="tolerance-grid">
+              {toleranceRanges.map((tolerance, index) => {
+                const { label, value } = tolerance;
+                return (
+                  <button
+                    key={`${label}-${index}`}
+                    className={cx('minute', { selected: minusTolerance === value })}
+                    style={{ gridArea: `a${index + 1}` }}
+                    onClick={() =>
+                      onToleranceChange({
+                        minusTolerance: value,
+                        plusTolerance: toleranceRanges[plusToleranceIndex].value,
+                      })
+                    }
+                  >
+                    {`-${label}`}
+                  </button>
+                );
+              })}
+            </div>
+          </div>
+
+          <div className="divider" />
+
+          <div className="time-picker-section">
+            <div className="tolerance-grid plus-tolerance">
+              {toleranceRanges.map((tolerance, index) => {
+                const { label, value } = tolerance;
+                return (
+                  <button
+                    key={`${label}-${index}`}
+                    className={cx('minute', { selected: plusTolerance === value })}
+                    style={{ gridArea: `a${index + 1}` }}
+                    onClick={() =>
+                      onToleranceChange({
+                        minusTolerance: toleranceRanges[minusToleranceIndex].value,
```

Comment:
You can keep the value from the input
```suggestion
                        minusTolerance,
```